File: src/java/ru/pavlenov/bio/utils/Str.java
package ru.pavlenov.bio.utils;

/**
 * ⓭ + 44
 * Какой сам? by Pavlenov Semen 01.07.14.
 */
public class Str {

    public static String prefix(String kmer, int k) {
        return kmer.substring(0, k);
    }

    public static String suffix(String kmer, int k) {
        return kmer.substring(kmer.length()-k, kmer.length());
    }

}

File: src/scala/ru/pavlenov/scala/utils/Str.scala
package ru.pavlenov.scala.utils

/**
 * ⓭ + 43
 * Какой сам? by Pavlenov Semen 05.07.14.
 */
object Str {

  /**
   * Подсчитываем кол-во символов в строке (символ => кол-во)
   *
   * @param str
   * @return
   */
  def counts(str: String): Map[Char, Int] = str.groupBy(c => c.toUpper).map(e => (e._1, e._2.length)).withDefaultValue(0)

  /**
   * Префикс строки, длины l
   * @param str
   * @param l
   * @return
   */
  def pref(str: String, l: Int): String = str.substring(0, l)

  /**
   * Суффикс строки длины l
   * @param str
   * @param l
   * @return
   */
  def suff(str: String, l: Int): String = str.substring(str.length - l, str.length)

}
